syntax = "proto2";
package avg_obj;

message PostUser {
    required string user_id = 1;
	optional string first_name = 2;
	optional string last_name = 3;
	enum UserType {
		FREE = 0;
		REGULAR = 1;
		PREMIUM = 2;
	}

	optional UserType user_type = 4 [default = FREE];
}

message MemeImage {

	required PostUser user = 1;
	required string title = 2;
	optional bytes content = 3;
	optional string top_string = 4;
	optional string bottom_string = 5;
	optional int64 likes = 6 [default = 0];
	optional int64 hates = 7 [default = 0];
	extensions 100 to 199;
}